### DISCUSSION TITLE
Fine-grained CRUD triggers

### DISCUSSION THREAD

**Discussion by manuelwehrli** [directus/directus#4812] [Feature Requests]
Could we have triggers on specific fields instead of a whole collection, maybe even with filter rules like the permissions have? It also would make sense to have a trigger for read event to have all CRUD actions covered.

**Comment by benhaynes** [comment]
You can do this on Event Hooks using custom logic, but Webhooks are a bit more "high-level". That said, you can use Event Hooks to create custom Webhooks too!

**Reply by manuelwehrli** [reply]
Ok, but why do webhooks not have read as trigger action?

**Reply by benhaynes** [reply]
I believe it was because that would be hit too often and flood the server with requests.

**Reply by manuelwehrli** [reply]
It should be in the hand of the developer to decide about that trade-off. When used sparely on few fields only, it will not lead to performance problems.

**Reply by sabinehoi** [reply]
Related: #3736. I think read API hooks would be more appropriate as they'd be lighter - then you can decide if you really wanna call a webhook from there. Also if read API hooks are implemented, we will be able to implement arbitrary permissions and filtering! Very useful for cases not covered by the json permissions system

**Comment by benhaynes** [comment]
This feature has been approved, and will be tracked here: https://github.com/directus/directus/issues/10164
